StreetDrop/ShareExtension/View/ShareViewController.swift

```diff
@@ -0,0 +1,846 @@
+        super.viewDidAppear(animated)
+        setupPlaceHolder()
+    }
+}
+
+private extension ShareViewController {
+    func bindAction() {
+        Observable.merge([
+            exitButton.rx.tap.asObservable(),
+            reSearchingMusicForSharingView.exitButtonEvent.asObservable()
+        ])
+        .bind(with: self) { owner, _ in
+            owner.extensionContext?.completeRequest(
+                returningItems: nil,
+                completionHandler: nil
+            )
+        }
+        .disposed(by: disposeBag)
+        
+        // 코멘트 플레이스홀더, 글자수라벨 설치, 줄 수 제한, 커멘트있을때만 드랍가능
+        commentTextView.rx.didBeginEditing
+            .subscribe { [weak self] element in
+                self?.removePlaceHolder()
+                self?.commentCountLabel.isHidden = false
+                self?.commentClearButton.isHidden = false
+                self?.commentClearButton.isEnabled = true
+                self?.commentView.layer.borderColor = UIColor.darkPrimary_25.cgColor
+            }.disposed(by: disposeBag)
+
+        commentTextView.rx.didEndEditing
+            .subscribe { [weak self] element in
+                self?.setupPlaceHolder()
+                self?.commentClearButton.isHidden = true
+                self?.commentClearButton.isEnabled = false
+                self?.commentView.layer.borderColor = UIColor.gray700.cgColor
+            }.disposed(by: disposeBag)
+
+        commentTextView.rx.didChange
+            .subscribe { [weak self] _ in
+                self?.checkMaxCount(max: 40)
+                self?.checkAvailableToDrop()
+            }.disposed(by: disposeBag)
+
+        commentTextView.rx.willBeginDragging
+            .subscribe { [weak self] _ in
+                self?.commentView.layer.borderColor = UIColor.darkPrimary_25.cgColor
+            }.disposed(by: disposeBag)
+
+        commentTextView.rx.didEndDragging
+            .subscribe { [weak self] _ in
+                self?.commentView.layer.borderColor = UIColor.gray700.cgColor
+            }.disposed(by: disposeBag)
+
+        commentTextView.rx.text.orEmpty
+            .asObservable()
+            .bind(with: self) { owner, text in
+                guard text != Constant.commentPlaceHolder,
+                      !text.isEmpty else { return }
+                owner.viewModel.comment = text
+                owner.commentCountLabel.text = "\(text.count)/40"
+                owner.commentView.layer.borderColor = text.count < 40
+                ? UIColor.darkPrimary_25.cgColor
+                : UIColor.systemCritical.cgColor
+            }.disposed(by: disposeBag)
+
+        commentClearButton.rx.tap
+            .bind { [weak self] in
+                self?.commentTextView.text = nil
+                self?.checkAvailableToDrop()
+            }.disposed(by: disposeBag)
+        
+        communityGuideButton.rx.tap
+            .bind { [weak self] in
+                guard let self = self else { return }
```

Comment:
옵셔널 제거 단축 syntax 로 써도 될거같아요~!
https://github.com/swiftlang/swift-evolution/blob/main/proposals/0345-if-let-shorthand.md
```suggestion
                guard let self else { return }
```